### PR TITLE
Update tutorial

### PR DESCRIPTION
**Pull request by piyush-garg**
This will update the tutorial to use two
lighweight applications which run fast
and concept of DAG and frontend/backend
application can be shown

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -125,11 +121,14 @@ When a task starts running, it starts a pod and runs each step sequentially in a
...
 Note that only the requirement for a git repository is declared on the task and not a specific git repository to be used. That allows tasks to be reusable for multiple pipelines and purposes. You can find more examples of reusable tasks in the [Tekton Catalog](https://github.com/tektoncd/catalog) and [OpenShift Catalog](https://github.com/openshift/pipelines-catalog) repositories.
 
-Install the `openshift-client` and `s2i-java` tasks from the catalog repository using `oc` or `kubectl`, which you will need for creating a pipeline in the next section:
+Install the `apply-manifests` and `update-deployment` tasks from the repository, `buildah` and `s2i-python-3` tasks from catalog repository
+using `oc` or `kubectl`, which you will need for creating a pipeline in the next section:
 
 ```bash
-$ oc create -f https://raw.githubusercontent.com/openshift/tektoncd-catalog/release-v0.7/openshift-client/openshift-client-task.yaml
-$ oc create -f https://raw.githubusercontent.com/openshift/pipelines-catalog/release-v0.7/s2i-java-8/s2i-java-8-task.yaml
+$ oc create -f https://raw.githubusercontent.com/openshift/pipelines-tutorial/master/voting-pipeline/update_deployment_task.yaml
```
Comment (siamaksade) [inline README.md]: can't we use the `openshift-cli` task instead of `update_deployment_task` and `apply_manifest_task`?

File: README.md [hunk 2]
```diff
@@ -138,69 +137,128 @@ You can take a look at the tasks you created using the [Tekton CLI](https://gith
...
 
 ## Create Pipeline
 
 A pipeline defines a number of tasks that should be executed and how they interact with each other via their inputs and outputs.
 
-In this tutorial, you will create a pipeline that takes the source code of the PetClinic application from GitHub and then builds and deploys it on OpenShift using [Source-to-Image (S2I)](https://docs.openshift.com/container-platform/4.1/builds/understanding-image-builds.html#build-strategy-s2i_understanding-image-builds).
+In this tutorial, you will create a pipeline that takes the source code of the application from GitHub and then builds and deploys it on OpenShift.
 
-<p align="center"><img src="images/pipeline-diagram.svg" width="700" alt="Pipeline Diagram" /></div>
+![Pipeline Diagram](images/pipeline-diagram.png)
```
Comment (siamaksade) [inline README.md:171]: Diagram doesn't match the screenshot of the pipeline. In one `build-ui` and `build-api` run in parallel, in the other one they are sequential.

Comment (piyush-garg) [reply README.md:171]: There is a bug in Devconsole diagram

File: demo [hunk 3]
```diff
@@ -55,22 +55,23 @@ demo.validate() {
   return 0
 }
 
-declare -r CATALOG_VERSION="release-v0.7"
-declare -r TEKTON_CATALOG="https://raw.githubusercontent.com/openshift/tektoncd-catalog/$CATALOG_VERSION"
-declare -r OPENSHIFT_CATALOG="https://raw.githubusercontent.com/openshift/pipelines-catalog/$CATALOG_VERSION"
+declare -r TEKTON_CATALOG_VERSION="release-next"
```
Comment (sthaha) [inline demo]: we shouldn't need this at all since the tasks are shipped in 0.8

File: demo [hunk 4]
```diff
@@ -55,22 +55,23 @@ demo.validate() {
...
-declare -r LOCAL_OC_TASK="$TASKS_DIR/oc.task.yaml"
+declare -r BUILDAH_TASK="$TEKTON_CATALOG/buildah/buildah.yaml"
+declare -r LOCAL_BUILDAH_TASK="$TASKS_DIR/oc.task.yaml"
 
-declare -r S2I_JAVA8_TASK="$OPENSHIFT_CATALOG/s2i-java-8/s2i-java-8-task.yaml"
-declare -r LOCAL_S2I_JAVA8_TASK="$TASKS_DIR/s2i-java8.task.yaml"
+declare -r S2I_PYTHON3_TASK="$OPENSHIFT_CATALOG/s2i-python-3/s2i-python-3-task.yaml"
+declare -r LOCAL_S2I_PYTHON3_TASK="$TASKS_DIR/s2i-python-3.task.yaml"
 
 demo.get-tasks() {
```
Comment (sthaha) [inline demo]: we should assume that operator installs this task

File: demo [hunk 5]
```diff
@@ -85,38 +86,45 @@ demo.setup() {
 
   demo.get-tasks
 
-  info "Apply petclinic resources"
-
-  sed -e "s|pipelines-tutorial|$NAMESPACE|g" petclinic/manifests.yaml | OC apply -f -
-
   info "Apply pipeline tasks"
   OC apply -f "$TASKS_DIR"
+  sed -e "s|pipelines-tutorial|$NAMESPACE|g" voting-pipeline/apply_manifest_task.yaml | OC apply -f -
```
Comment (sthaha) [inline demo]: shouldn't the pipeline apply the manifests from the projects instead?

File: voting-pipeline/pipeline.yaml [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+            resource: api-image
+      params:
+        - name: deployment
+          value: "voting-api"
+      runAfter:
+        - apply-api-manifests
+
+    - name: build-ui
+      taskRef:
+        name: s2i-python-3
```
Comment (sthaha) [inline voting-pipeline/pipeline.yaml]: should use the clusterTask instead

File: README.md (hunk 1 above)
Comment (piyush-garg) [reply README.md]: no, we need to pass the git resource to apply the manifests from individual repo's

File: README.md (hunk 2 above)
Comment (piyush-garg) [reply README.md:171]: Even in the latest dev console, it is not showing properly

bug-flow

**Comment by siamaksade** [comment]
Please create a bug on dev console JIRA with the example pipeline

**Comment by PierreR** [comment]
It would be nice to have an example that uses a `Deployment` instead of a `DeploymentConfig`. The [openshift 4.x documentation](https://docs.openshift.com/container-platform/4.2/applications/deployments/what-deployments-are.html#deployments-comparing-deploymentconfigs_what-deployments-are) states:
> It is recommended to use `Deployments` unless you need a specific feature or behavior provided by `DeploymentConfigs` footnote:[https://docs.openshift.com/container-platform/4.2/applications/deployments/what-deployments-are.html]. Deployments do not support (yet) automatically rolling back to the last successfully deployed ReplicaSet in case of a failure.

Does openshift pipelines requires the usage of `DeploymentConfig` ?

**Comment by PierreR** [comment]
Also I believe the sentence:
>You might have noticed that there are no references to the PetClinic Git repository and its image in the registry. 

is now out-of-sync in the README.

**Comment by sthaha** [comment]
> It would be nice to have an example that uses a `Deployment` instead of a `DeploymentConfig`. The [openshift 4.x documentation](https://docs.openshift.com/container-platform/4.2/applications/deployments/what-deployments-are.html#deployments-comparing-deploymentconfigs_what-deployments-are) states:

@PierreR what section in the PR are you referring to?

**Pull request merged**